core/src/androidMain/kotlin/Connection.kt

```diff
@@ -24,20 +22,13 @@ public class OutOfOrderGattCallbackException internal constructor(
 private val GattSuccess = GattStatus(GATT_SUCCESS)
 
 internal class Connection(
-    scope: CoroutineScope,
+    private val scope: CoroutineScope,
     internal val bluetoothGatt: BluetoothGatt,
     internal val dispatcher: CoroutineDispatcher,
     private val callback: Callback,
     logging: Logging,
 ) {
 
-    private val job = Job(scope.coroutineContext[Job]).apply {
-        invokeOnCompletion {
-            bluetoothGatt.close()
```

Comment:
Not needed (was never needed). We `close` the GATT object in the Callback. This was an unnecessary extra call to `close`.